src/Model/ModelManager.php

```diff
@@ -339,12 +339,6 @@ public function getIdentifierValues($entity)
                 continue;
             }
 
-            if (method_exists($value, '__toString')) {
-                $identifiers[] = (string) $value;
-
-                continue;
-            }
-
```

Comment:
IIRC, optimised UUID failed (which shows in the travis) as it meets the below `if($type)` condition. Meaning, `__toString` likely needs to place before the `$type` check.

I did a little bit digging on the doctrine composite key a few weeks ago and notice that it is resolved by the `$metadata->getIdentifierValues(...)` below.

Maybe, for a quick win to serve all known use cases, we probably just need to check with the $metadata condition.

```php
if (method_exists($value, '__toString') && 0 === \count($metadata->getIdentifierValues(...)) {
    // ...
}
```

Reply:
(I didn't have time to test it and therefore didn't PR)

Reply:
I make test with UuidBinaryOrderedTimeType of Ramsey and with the last solution is working. 
We have to test if the object for the id field is a binary type in DB or not. If it is binary, it has no sense that we use "convertToDatabaseValue" due to we get a binary value. We need use convertToPHPValue instead and cast it to string.
With this we get solution for both case (ID composite and Binary object ID)

Reply:
Doing binary type check makes sense for the UuidBinaryOrderedTimeType. However this approach seems not so generalised. Object as an ID may not always persist as binary. For example, UuidType persists as string (https://github.com/ramsey/uuid-doctrine/blob/master/src/UuidType.php#L91) and the current proposal may break it.

Reply:
No @torreytsui, there is no that problem. If you use UuidType like Ramsey's bundle, the database type of this object is not binary. In this case, it's used $type->convertToDatabaseValue.
It doesn't make sense to check if the object has the __toString method when the value is already stored in DB in a format that can be "used" as an index in Sonata Admin.
This solution cover all the case of entity's id that you have.

Reply:
Oh, I see. I overlooked it, it makes more sense to me now. Basically, we assume that all non-binary type db values can be used directly.